test/test_transformer.cc

```diff
@@ -518,3 +518,211 @@ TEST(Transformer, Codegen) {
+  // id for mark nodes.
+  std::string markerTile = "first level tiling - tile";
+  isl::id idTile = isl::id::alloc(node.get_ctx(), markerTile, nullptr);
+  std::string markerPoint = "first level tiling - point";
+  isl::id idPoint = isl::id::alloc(node.get_ctx(), markerPoint, nullptr);
+
+  // clang-format off
+  ScheduleNodeBuilder macroKernel = 
+    mark(idTile,
+      band(
+        [&node, &tileSchedule, &dimOutNum]() {
+          auto newPartialSchedule = permuteDims(tileSchedule, dimOutNum - 2, dimOutNum - 1);
+          newPartialSchedule = permuteDims(newPartialSchedule, dimOutNum - 3, dimOutNum - 1);
+          return newPartialSchedule;},
+        mark(idPoint,
+	  band(pointSchedule))));
+  // clang-format on
+
+  if(matched) {
+    node = rebuild(node, macroKernel);
+    root = node.root();
+    //std::cout << root.to_str() << std::endl;
+  }
+
+  // match micro-kernel
+  auto matcherMicroKernel = [&]() {
+    using namespace matchers;
+    // clang-format off
+      return band(node,leaf());
+    // clang-format on
+  }();
+
+  matched = matched && 
+    ScheduleNodeMatcher::isMatching(matcherMicroKernel, root.child(0).child(0).child(0).child(0)); 
+  EXPECT_TRUE(matched);
+  
+  // create micro-kernel
+  // tiling Mr = 4 and Nr = 8 and 1.
+  tiledNode = tile(node, {4,8,1});
+  tileSchedule = tiledNode.band_get_partial_schedule();
+  pointSchedule = tiledNode.child(0).band_get_partial_schedule();
+  // id for mark nodes.
+  markerTile = "register tiling - tile";
+  idTile = isl::id::alloc(node.get_ctx(), markerTile, nullptr);
+  markerPoint = "register tiling - point";
+  idPoint = isl::id::alloc(node.get_ctx(), markerPoint, nullptr);
+
+  // clang-format off  
+  ScheduleNodeBuilder microKernel = 
+    mark(idTile,
+      band(
+        [&node, &tileSchedule]() {
+          auto newPartialSchedule = permuteDims(tileSchedule, 0, 1);
+          return newPartialSchedule;},
+        mark(idPoint,
+          band(
+            [&node, &pointSchedule]() {
+              auto descr = BandDescriptor(pointSchedule);
+              descr.astOptions = isl::union_set(node.get_ctx(), "{unroll[x]}");
+  	      return descr;}))));
+  // clang-format on
+
+  if(matched) {
+    node = rebuild(node, microKernel);
+    root = node.root();
+    //std::cout << root.to_str() << std::endl;
+  }
+/*
+  // isolate full-tile from partial tile
+  isl::schedule_node leafNode;
+  auto matcherMicroKernelAfterTiling = [&]() {
+    using namespace matchers;
+    // clang-format off
+      return band(node, leaf(leafNode));
+    // clang-format on
+  }();
+
+  matched = matched &&
+    ScheduleNodeMatcher::isMatching(matcherMicroKernelAfterTiling, 
+      root.child(0).child(0).child(0).child(0).child(0).child(0).child(0));
+  EXPECT_TRUE(matched);
+*/    
+  std::cout << petScop.codegen() << std::endl;
+  petScop.schedule() = root.get_schedule();
+  std::cout << petScop.codegen() << std::endl;
```

Comment:
How about you check for _some_ pattern in the code instead of just dumping it.  Like the new loops are indeed produced or something?